src/QMCWaveFunctions/SPOSetBuilder.cpp

```diff
@@ -98,6 +98,9 @@ std::unique_ptr<SPOSet> SPOSetBuilder::createSPOSet(xmlNodePtr cur)
     // create sposet with rotation
     auto& sposet_ref = *sposet;
     app_log() << "  SPOSet " << sposet_ref.getName() << " is optimizable\n";
+    if (!sposet_ref.isRotationSupported())
```

Comment:
Why not throw a UniformCommunicateError here?

Reply:
At this point, I can already safely stop the code and mpi communicator is available here. I think UniformCommunicateError is better suited in classes without mpi communicator.
returning UniformCommunicateError may got stuck if another exception happens in destruction due to earlier issues. So I feel better to stop early whenever possible.

Reply:
Seems reasonable.